test/test.js

```diff
@@ -1,119 +1,112 @@
+var extract = require('../')
+var fs = require('fs')
 var os = require('os')
 var path = require('path')
 var rimraf = require('rimraf')
+var temp = require('temp').track()
 var test = require('tape')
-var fs = require('fs')
-var extract = require('../')
 
-var sourceA = path.join(__dirname, 'cats.zip')
-var targetA = path.join(os.tmpdir(), 'cat-extract-test')
-var resultsA = path.join(targetA, 'cats')
+var catsZip = path.join(__dirname, 'cats.zip')
+var githubZip = path.join(__dirname, 'github.zip')
+var subdirZip = path.join(__dirname, 'file-in-subdir-without-subdir-entry.zip')
+var symlinkDestZip = path.join(__dirname, 'symlink-dest.zip')
+var symlinkZip = path.join(__dirname, 'symlink.zip')
 
-var sourceB = path.join(__dirname, 'github.zip')
-var targetB = os.tmpdir()
-var resultsB = path.join(targetB, 'extract-zip-master')
+var relativeTarget = './cats'
 
-var sourceC = path.join(__dirname, 'symlink.zip')
-var targetC = os.tmpdir()
-
-var sourceD = path.join(__dirname, 'cats.zip')
-var targetD = './cats'
-
-var sourceE = path.join(__dirname, 'symlink-dest.zip')
-var targetE = os.tmpdir()
-var resultsE = path.join(targetB, 'symlink-dest')
-
-test('extract cat zip', function (t) {
-  rimraf.sync(targetA)
+function mkdtemp (suffix, callback) {
+  temp.mkdir({prefix: 'extract-zip', suffix: suffix}, function (err, dirPath) {
+    if (err) throw err
```

Comment:
This should probably be `t.same(err, null)` the throw will be lost in a long callback chain